lib/request.js

```diff
@@ -1,29 +1,20 @@
 'use strict'
 
 const { Readable } = require('readable-stream')
-const qs = require('querystring')
 const util = require('util')
-const url = require('url')
-
-function parseURL (options) {
-  const parsedURL = url.parse(
-    typeof options.url === 'object'
-      ? url.format(options.url)
-      : options.url
-  )
-
-  // Returns non-empty string only non-empty object
-  if (qs.stringify(options.query)) {
-    const original = qs.parse(parsedURL.query)
-    const combined = Object.assign(original, options.query)
-
-    const query = qs.stringify(combined)
-    if (query) {
-      parsedURL.path = parsedURL.pathname + '?' + query
-    }
-  }
 
-  return parsedURL
+const parseURL = require('./parseURL')
+
+/**
+ * Get hostname:port
+ *
+ * @param {URL} parsedURL
+ * @return {String}
+ */
+function hostHeaderFromURL (parsedURL) {
+  return parsedURL.port
```

Comment:
What are you checking here?
The existence of the `port` value?
If so, I would add a stricter check.

Reply:
The new URL's `port` will be empty when using default ports:

```javascript
new URL('http://localhost').port === ''
new URL('http://localhost:80').port === ''
new URL('https://localhost').port === ''
new URL('https://localhost:443').port === ''
```